src/Installers/Windows/SharedFrameworkBundle/Bundle.wxs

```diff
@@ -24,6 +26,54 @@
         <?endif?>
         <?endif?>
 
+        <!-- Ensure upgrades from 3.1.0 thru 3.1.8 for x64. -->
+        <?if $(var.Version)=3.1.0.0?>
```

Comment:
Has the value of Version changed at all with moving to the non-stable versions? Just wondering in case it throws this off. I wander if to be safe the check here should be to see whether it's building for 3.1

Reply:
Oh, yes, `Version` will be different with each patch - are these msbuild properties? or DefineConstants from wix.targets? If the former we could use `AspNetCoreMajorMinorVersion`, if the latter we could use `MajorVersion` and `MinorVersion`

Reply:
They are either MSBUILD properties that we just pass to WiX (through the DefineConstants), or they might be special ones we build up based off existing properties and repurpose for WiX, but there should be a DefineConstant in either a wixproj or one of the wix props/targets files I think

Reply:
https://github.com/dotnet/aspnetcore/pull/26064/commits/fc60a8acbdfaf03933e48df4f1e7d5bcabecd088

MajorVersion and MinorVersion are defined in `wix.targets`